### DISCUSSION TITLE
Link to Text Fragment-  a special link to the currently selected text on a page via the context menu

### DISCUSSION THREAD

**Discussion by Raytle** [gildas-lormeau/SingleFile#771] [Ideas]
With the following Google Chrome extension installed [Link to Text Fragment](https://chrome.google.com/webstore/detail/link-to-text-fragment/pbcodcjpfjdpcineamnnmbkkmkdpajjg?hl=en) one can [Boldly link where no one has linked before: Text Fragments](https://web.dev/text-fragments/).

For example, I can link to...

https://en.wiktionary.org/wiki/the_proof_of_the_pudding_is_in_the_eating#:~:text=You%20can,the%20pudding so that when I click on that link I see the following—in Google Chrome but **not** in Firefox on Lubuntu 20.04 LTS—(**without** the green line encircling the text and **without** the yellow arrow)...

![image](https://user-images.githubusercontent.com/81956552/132252639-f473ac2b-e64b-4761-9bcb-bf308dc6b605.png)

[Link to Text Fragment](https://chrome.google.com/webstore/detail/link-to-text-fragment/pbcodcjpfjdpcineamnnmbkkmkdpajjg?hl=en) is a...

"[b]rowser extension that allows for linking to arbitrary text on a page.

This extension allows for easily creating a special link to the currently selected text on a page via the context menu. When opening such a special link, a compatible browser will scroll the selected text into view and highlight it."

I would like a similar feature in SingleFile.

I suppose the use case I have in mind is obvious, but to ensure you understand me, when I save a webpage with SingleFile I am normally most interested in a particular part of that webpage. Therefore, I would like to be able to link directly to that portion of the webpage I am most interested in yet still be able to read other portions of the webpage which I am less interested in.

**Comment by gildas-lormeau** [comment]
Depending on your needs, it might already work by enabling "Allow access to file URLs" in the extension page of Link to Text Fragment, see chrome://extensions/?id=pbcodcjpfjdpcineamnnmbkkmkdpajjg

**Comment by Raytle** [comment]
Thanks for replying quickly to me. I enabled "Allow access to file URLs." Then I saved a link with Link to Text Fragment. I didn't notice any difference. I found this [What is "Allow access to file URLs" in Chrome?](https://support.logmeininc.com/lastpass/help/what-is-allow-access-to-file-urls-in-chrome). 

Did you test it yourself?

**Comment by gildas-lormeau** [comment]
I did not do the same test:
1. I saved a page with SingleFile
2. I selected a part of the saved page a created a link with Link to Text Fragment
3. I opened the link in a new tab and it worked as expected

**Comment by Raytle** [comment]
Thanks for the information and thanks for once again replying quickly to me.

1. I saved a page with SingleFile using **Firefox** on Lubuntu 20.04 LTS.
2. Then I opened that page with **Google Chrome** on Lubuntu 20.04 LTS.
3. Then I selected some text on that page yet I was **unable** to create a link with Link to Text Fragment.

However, I think that problem might be moot because I suppose I might solve my underlying problem (that is, my fundamental problem) another way. Are you familiar with  [How to Bookmark Specific Locations in a Google Docs File](https://www.howtogeek.com/397357/how-to-bookmark-specific-locations-in-a-google-docs-file/)?

I suppose I might..

1. upload every page I create with SingleFile to my Google Drive account, and then
2. transform each of those SingleFiles into a Google Doc, and then
3. find a location I wanted to link to within each of those Google Docs, and then
4. choose `Insert –> Bookmark`, and then
3. copy each of those URLs (each bookmark in a Google Doc has a URL), and then
4. use those URLs in other documents to create hyperlinks to particular sections of my Google Docs (which had formerly been SingleFiles).

In other words, the bookmark feature in a Google Docs was designed for the specific purpose I have in mind. Also, by using Google Docs I would be able to easily add all sort of notes and comments to documents. Sure, of course I could use a wiki instead of Google Docs. But I would prefer to use Google Docs because it has outline mode, styles, voice dictation, and—very importantly for me—Google Docs are trivial for most non-technical people to use (including female artists and female designers). Also, Google Docs has been extremely reliable for me for over a decade.

However, my "Google Docs solution" would create a couple of problems. 

First, I would need to manually reformat most of SingleFile webpages that I had uploaded to Google Drive and transformed into a Google Doc to make each of them look merely pretty good in Google Docs. By pretty good, I mean not as good as those files had originally looked when they had been ordinary web pages. This is **not** a theoretical concern. Rather this a practical problem which I have experienced. I have found that I normally need to spend a few minutes manually reformatting each of these Google Docs so that they look merely pretty good. But this is not a deal-breaker. Instead, it is a minor chore.

Second, although of course I could use [Google Takeout](https://takeout.google.com/) to backup (download) my files from my Google Drive account to the solid state drive on my laptop, if Google were to cancel my Google Drive account, then all the aforementioned bookmarks would immediately experience [link rot](https://en.wikipedia.org/wiki/Link_rot) (stop working). In other words, although I would presumably have a backed up all of my files, which had been on my Google Drive account, to my local drive (that is, to my solid state drive inside of my laptop) the links **between** those files would **fail** to work in case I were to upload those files to a **new** Google Drive account.

Obviously these two problems are well beyond the scope of SingleFile. Nevertheless, do you have any suggestions for me for solving either of the two problems listed immediately above?

I want to reiterate: I've used and tested many wikis, yet I strongly prefer to use Google Drive (most Google Docs) instead of a wiki for this application.
